src/base/emu-i386/simx86/interp.c

```diff
@@ -967,6 +955,36 @@ unsigned int Sim_helper(unsigned int mem_ref, unsigned int data, int mode,
 			}
 			EFLAGS &= ~EFLAGS_VIF;
 			break;
+/*fb*/	case STI:
+			if (V86MODE()) {    /* traps always (Intel man) */
+				/* virtual-8086 monitor */
+				if (IOPL==3)
+				    EFLAGS |= EFLAGS_IF;
+				else
+				    EFLAGS |= EFLAGS_VIF;
+				if (vm86s.regs.eflags & VIP) {
+				    if (debug_level('e')>1)
+					e_printf("Return for STI fl=%08x\n",
+					    EFLAGS);
+				    TheCPU.err2=EXCP_STISIGNAL;
+				}
+			}
+			else {
+			    if (REALMODE() || (CPL <= IOPL) || (IOPL==3)) {
+				EFLAGS |= EFLAGS_IF;
+			    }
+			    else {
+				if (debug_level('e')>2) e_printf("Virtual DPMI STI\n");
+				EFLAGS |= EFLAGS_VIF;
+			    }
+			    if (pic_pending()) {
```

Comment:
Would it make sense to
change this to:
```
if ((!PVI && pic_pending())|| (PVI && isset_VIP())
```
?

Reply:
that would make sense but it would be new functionality, not just moving it around vs `devel`, as we're not really using or testing PVI at the moment.